File: Comp/one/one.py
def max_revenue(tuitions):
  # Sort the tuitions in non-ascending order
  tuitions.sort(reverse=True)

  # Set the initial revenue to 0 and the initial number of attendees to 0
  revenue = 0
  attendees = 0

  # Iterate through the tuitions
  for tuition in tuitions:
    # Add the tuition to the revenue and increment the number of attendees
    revenue += tuition
    attendees += 1

    # If the revenue is less than the number of attendees times the tuition,
    # then the current tuition is not the optimal solution
    if revenue < attendees * tuition:
      break

  # Return the total revenue and the optimal tuition
  return revenue, tuition

# Test the function with the sample input
tuitions = [1, 6, 4, 6]
print(max_revenue(tuitions))  # Output: (12, 4)

File: Comp/one/one.java
import java.util.Arrays;
import java.util.Scanner;

public class one {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);

    // Read in the number of cows
    int n = scanner.nextInt();

    // Read in the maximum tuitions that each cow is willing to pay
    long[] tuitions = new long[n];
    for (int i = 0; i < n; i++) {
      tuitions[i] = scanner.nextInt();
    }

    // Sort the tuitions in ascending order
    Arrays.sort(tuitions);

    // Initialize the maximum amount of money that can be made and the optimal tuition to charge
    long maxMoney = 0;
    long optimalTuition = 0;

    // Iterate through the tuitions and calculate the maximum amount of money that can be made
    // and the optimal tuition to charge
    for (int i = 0; i < n; i++) {
      long tuition = tuitions[i] * (n - i);
      if (tuition > maxMoney) {
        maxMoney = tuition;
        optimalTuition = tuitions[i];
      }
    }

    // Print the results
    System.out.println(maxMoney + " " + optimalTuition);
  }
}
